### PR TITLE
Reply button is not removed when deleting a code review comment

### LINKED ISSUE #13764: Reply button is not removed when deleting a code review comment
- Gitea version (or commit ref): 1.12.6
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (provide example URL)
  - [ ] No

## Description

When deleting a code review comment, the "Reply" and "Resolve Conversation" buttons remain on the page. They are gone when you refresh the page. When you delete a resolved reply comment, that part of the UI also remains on the page.

## Screenshots

I delete this comment:

image

The "Reply" and "Resolve Conversation" buttons remain on the page:

image

When I delete a resolved comment:

image

The "Reply", "Unresolve Conversation" buttons remain, and also the "Hide resolved" placeholder remains.

image

### PR DESCRIPTION
**Pull request by jpraet**
Fixes #13764

This adds a conversation-wrapper div around a code review comment thread. When the last comment in the conversation gets deleted, the conversation-wrapper now also gets deleted, removing the reply / (un)resolve buttons and resolved conversation wrapper.

I also extracted the conversation, which was duplicated in box.tmpl (left+right) and section_unified.tmpl, into a separate template.

### CODE REVIEW

File: web_src/js/index.js [hunk 1]
```diff
@@ -1094,7 +1094,11 @@ async function initRepository() {
         $.post($this.data('url'), {
           _csrf: csrf
         }).done(() => {
+          const $conversationHolder = $this.closest('.conversation-holder')
```
Comment (mrsdizzie) [inline web_src/js/index.js]:
```suggestion
          const $conversationHolder = $this.closest('.conversation-holder');
```

**Review by silverwind — APPROVED** [review]

**Comment by 6543** [comment]
@jpraet by the way: if you allow maintainers to edit pulls - we could update pull's too :)

**Review by 6543 — APPROVED** [review]

**Comment by lafriks** [comment]
please merge latest changes

**Comment by jpraet** [comment]
> @jpraet by the way: if you allow maintainers to edit pulls - we could update pull's too :)

For some reason the "Allow edits from maintainers" checkbox is not showing up for me.

**Comment by lafriks** [comment]
yeah, I think it is available only when creating PR :(

**Comment by silverwind** [comment]
Very nice work, always great to see template deduplications which we are very guilty of 😉

**Comment by 6543** [comment]
we should backport the fix - but I'm not sure if we should also backport template dedub?

**Comment by silverwind** [comment]
This also has a dependency on the new `mergeinto` helper which I think is not in 1.13.

**Comment by 6543** [comment]
@jpraet can you create a backport to 1.13 without the dedublication refactor ?

**Comment by jpraet** [comment]
@6543 ok

**Pull request merged**